doc/rados/operations/placement-groups.rst

```diff
@@ -700,134 +705,146 @@ For example:
 
 .. prompt:: bash #
 
    ceph pg cancel-force-recovery {pg-id} [{pg-id #2}] [{pg-id #3} ...]
    ceph pg cancel-force-backfill {pg-id} [{pg-id #2}] [{pg-id #3} ...]
 
-This will remove "force" flag from those PGs and they will be processed
-in default order. Again, this doesn't affect currently processed placement
-group, only those that are still queued.
+These commands remove the ``force`` flag from the specified PGs, so that the
+PGs will be processed in their default order. As in the case of adding the
+``force`` flag, this affects only those PGs that are still queued but does not
+affect currently processed PGs.
 
-The "force" flag is cleared automatically after recovery or backfill of group
-is done.
+The ``force`` flag is cleared automatically after the recovery or backfill of
+the PGs is complete.
 
-Similarly, you may use the following commands to force Ceph to perform recovery
-or backfill on all placement groups from a specified pool first:
+Similarly, to instruct Ceph to prioritize all PGs from a specified pool (that
+is, to perform recovery or backfill on those PGs first), run one or both of the
+following commands:
 
 .. prompt:: bash #
 
    ceph osd pool force-recovery {pool-name}
    ceph osd pool force-backfill {pool-name}
 
-or:
+These commands can also be cancelled. To revert to the default order, run one
+or both of the following commands:
 
 .. prompt:: bash #
 
    ceph osd pool cancel-force-recovery {pool-name}
    ceph osd pool cancel-force-backfill {pool-name}
 
-to restore to the default recovery or backfill priority if you change your mind.
-
-Note that these commands could possibly break the ordering of Ceph's internal
-priority computations, so use them with caution!
-Especially, if you have multiple pools that are currently sharing the same
-underlying OSDs, and some particular pools hold data more important than others,
-we recommend you use the following command to re-arrange all pools's
-recovery/backfill priority in a better order:
+.. note:: These commands can break the order of Ceph's internal priority
+   computations, so use them with caution! If you have multiple pools that are
+   currently sharing the same underlying OSDs, and if the data held by certain
+   pools is more important than the data held by other pools, then we recommend
+   that you run a command of the following form to arrange a custom
+   recovery/backfill priority for all pools:
 
 .. prompt:: bash #
 
    ceph osd pool set {pool-name} recovery_priority {value}
 
-For example, if you have 10 pools you could make the most important one priority 10,
-next 9, etc. Or you could leave most pools alone and have say 3 important pools
-all priority 1 or priorities 3, 2, 1 respectively.
+For example, if you have 10 pools you could make the most important one
+priority 10, next 9, etc. Or you could leave most pools alone and have say 3
+important pools all priority 1 or priorities 3, 2, 1 respectively.
```

Comment:
@colemitchell: Didn't you have some notes about this?

Reply:
@zdover23 - Yeah, the question is whether the low-priority to high-priority scale is a fixed scale of 1 to 10 (as is suggested by https://access.redhat.com/documentation/en-us/red_hat_ceph_storage/4/html/administration_guide/monitoring-a-ceph-storage-cluster), or instead a scale whose maximum is determined by the number of pools (e.g., if there are seventeen pools then it goes from 1 to 17).

Reply:
I had no idea we could do this.  Maybe it's relatively new.  With OSD op priority, the values were relative to each other, but this is at least partly independent.  Looking at the RHCS document linked above, I can see where one might infer that it's a scale of [1-10], but I think that's a co-incidence.

Use the source, Luke.  cf. `PeeringState.cc`:


```          ss << "pool recovery_priority must be between " << OSD_POOL_PRIORITY_MIN
             << " and " << OSD_POOL_PRIORITY_MAX;```
```    if (prio > 0 && max_prio > OSD_POOL_PRIORITY_MAX) { // Likely scenario
      // Scaled priority range 0 to OSD_POOL_PRIORITY_MAX
      n = (float)prio / max_prio * OSD_POOL_PRIORITY_MAX;
    } else if (prio < 0 && min_prio < OSD_POOL_PRIORITY_MIN) {
      // Scaled  priority range OSD_POOL_PRIORITY_MIN to 0
      n = (float)prio / min_prio * OSD_POOL_PRIORITY_MIN;
    } else {
      continue;
    }```
  
```    // Adjust with pool's recovery priority
    int64_t pool_recovery_priority = 0;
    pool.info.opts.get(pool_opts_t::RECOVERY_PRIORITY, &pool_recovery_priority);

    ret = clamp_recovery_priority(ret, pool_recovery_priority, max_prio_map[base]);```

So I won't testify under oath (or re Underoath, who are overrated) but I _think_ the pool priority acts like a weight applied after smaller-granularity calculations.  And thus that the priorities of a set of pools are, like OSD op priorities, ordinally relative to each other independent of the number of pools.

Given the values in `osd_types.h`, I additionally think that a higher number means higher priority; in computing or life in general sometimes we see the opposite.  

So as to what we do with the text, I suggest `s/all priority 1//`.

Separately, if we wish to avoid implying that values are bounded by the number of pools, we might change line 820 to be something like "priority 6, the next priority 5, etc."

Reply:
I've altered the text so that it reflects the idea that "greater value means higher priority" and I have chosen example values of greater than ten to make it clear that this is not a decimal scale.